libraries/AP_RangeFinder/AP_RangeFinder_Benewake_TFMiniPlus.cpp

```diff
@@ -102,6 +104,8 @@ bool AP_RangeFinder_Benewake_TFMiniPlus::init()
     if (val[5] * 10000 + val[4] * 100 + val[3] < 20003) {
         hal.console->printf(DRIVER ": minimum required FW version 2.0.3, but version %u.%u.%u found\n",
                             val[5], val[4], val[3]);
+        gcs().send_text(MAV_SEVERITY_CRITICAL, "TFMini: FW ver %u.%u.%u (need>=2.0.3)",
```

Comment:
This is being called with a semaphore held.  This is actually really not a good idea - we want to cross as little code as possible when holding a semaphore - and `gcs().send_text()` isn't it :-)

Could you stash the three values in some variables and detect the version problem outside the semaphore-held part, please?